### ISSUE TITLE
Conversion of tensor shape values to None in a keras model

### ISSUE THREAD

**Issue by AdityaNikhil** [tensorflow/tensorflow#50275]
## Suppose I have a simple keras model as follows, 

    Model: "sequential_1"
    _________________________________________________________________
    Layer (type)                 Output Shape              Param #   
    =================================================================
    conv2d (Conv2D)              (None, 16, 16, 8)         224       
    _________________________________________________________________
    conv2d_1                     (None, 7, 7, 8)           584       
    _________________________________________________________________
    conv2d_2                     (None, 3, 3, 2)           584       
    =================================================================
    Total params: 1,392
    Trainable params: 1,392
    Non-trainable params: 0

## Now I want the final output layer shape as follows,

    Model: "sequential_1"
    _________________________________________________________________
    Layer (type)                 Output Shape              Param #   
    =================================================================
    conv2d (Conv2D)              (None, 16, 16, 8)         224       
    _________________________________________________________________
    conv2d_1                     (None, 7, 7, 8)           584       
    _________________________________________________________________
    conv2d_2                     (None, None, None, 2)     584       
    =================================================================
    Total params: 1,392
    Trainable params: 1,392
    Non-trainable params: 0

## Code

    m = Sequential()
    
    m.add(layers.InputLayer(input_shape=(32, 32, 1)))
    m.add(Conv2D(8, (3,3), padding='same', strides=2))
    m.add(Conv2D(8, (3,3), strides=2))
    m.add(Conv2D(2, (3,3), strides=2))

## Details
> I hope you got an understanding of the issue.
> Please try to make your own versions of the code. This is just a demo code.

I am working on image colorization problem where we input an image with one color channel and get output with 2 color channels.
The original code is huge and different from any codes on the internet. So please don't be concerned about suggesting me any image colorization codes. 

I really appreciate it if you can help me with this issue.

**Comment by tilakrayal** [comment]
@AdityaNikhil ,

In order to expedite the trouble-shooting process, could you please provide the tensorflow version, complete code and dataset to reproduce the issue reported here.

Thanks!

**Comment by AdityaNikhil** [comment]
@tilakrayal 

Here's the code I am working on. I've also mentioned what kind of architecture I expect at the end. 
Please look into this thoroughly. It'll be a huge help for my research. 
https://colab.research.google.com/drive/1Tj1hRejjSjd5lq-Zxm15_LhdmKXOqLwY?usp=sharing

Thannk you

**Comment by tilakrayal** [comment]
@AdityaNikhil ,

Can you please take a look at links with similar error.Hope it helps.[Link1](https://github.com/tensorflow/tensorflow/issues/48513),[Link2](https://datascience.stackexchange.com/questions/64072/input-0-is-incompatible-with-layer-conv2d-2-expected-ndim-4-found-ndim-3-i-get),[Link3](https://stackoverflow.com/questions/47665391/keras-valueerror-input-0-is-incompatible-with-layer-conv2d-1-expected-ndim-4).
Thanks!

**Comment by AdityaNikhil** [comment]
@tilakrayal 
Thank you for the links but still, they don't fix the issue.

**Comment by tilakrayal** [comment]
@ymodak ,

I was able to reproduce the issue in tf [v2.4](https://colab.research.google.com/gist/tilakrayal/10912edf8c5533b186dc9ed7e5bfe993/2-4colorization_using_transformers.ipynb),[v2.5](https://colab.research.google.com/gist/tilakrayal/c286bd2249ee22486f70cd3f594494f8/2-5colorization_using_transformers.ipynb) and [nightly](https://colab.research.google.com/gist/tilakrayal/69f6786833831094d854def11ebdceeb/tf-nightlycolorization_using_transformers.ipynb).Please find the gist here.

**Comment by ymodak** [comment]
This question is better asked on [TensorFlow Forum](https://discuss.tensorflow.org/) since it is not a bug or feature request. There is also a larger community that reads questions there.
Thanks!
